analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotUseDateTimeNow.cs

```diff
@@ -0,0 +1,32 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class DoNotUseDateTimeNow : DoNotUseDateTimeNowBase<SyntaxKind>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+    protected override bool IsInsideNameOf(SyntaxNode node) =>
+        node.Ancestors()
```

Comment:
Because of this check if the user creates a method called nameof that takes a DateTime we will not raise. Either cover it or if it is expensive cover it as a FN (it is an edge case)

Reply:
In VB, the syntax is somewhat different (there's a specific syntax class for `nameof`), so it's going to be a FN in C#, but Noncompliant in VB.